File: coffeetime-parent/RubyGun/src/29_io.rb
unless Dir.exist? 'files'
  Dir.mkdir 'files'
end

Dir.foreach('/') do |entry|
  puts entry
end

puts File.exist? 'files/test.txt'

Dir.chdir('files')
file = File.new('test.txt', 'w')
file.syswrite('hello world')
file.write('foobar')
file.print("world\n")
file.puts('bye')
file.close

File.open('test.txt', 'r') do |file|
  puts "read: #{file.read}"
end

lines = IO.readlines('test.txt')
puts "line: #{lines[0]}"

IO.foreach('test.txt') { |block| puts "each: #{block}" }
file = File.new('test.txt', 'r')
file.each_byte {|ch| putc ch}

File: coffeetime-parent/Python3Gun/src/29_io.py
import os

if not os.path.exists('files'):
    os.mkdir('files')

entries = os.listdir('files')
for entry in entries:
    print(entry)

print(os.path.isfile('files/test.txt'))

os.chdir('files')

file = open('test.txt', 'w', encoding='UTF-8')
file.write('foobar')
file.write('world\n')
file.write('bye')
file.close()

file = open('test.txt', 'r')
print('read:', file.read())

# Try-Resource
try:
    with open('test.txt', 'r') as f:
        for line in f:
            print(line)
except OSError:
    print('OSError occurs')
else:
    print('end read')

# Current Position
print(file.tell())

file.seek(0)
lines = list(file)
print(lines)

file.seek(0)
for line in file:
    print(line)
file.close()
